File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Announcement.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "announcements", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "announcements_unique", columnNames = {"announcement_name"})
})
public class Announcement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "announcement_id", nullable = false)
    private Long id;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @NotNull
    @Column(name = "message", nullable = false, length = Integer.MAX_VALUE)
    private String message;

    @NotNull
    @Column(name = "announcement_name", nullable = false, length = Integer.MAX_VALUE)
    private String announcementName;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "session_id")
    private Session session;

}

File: umbrella-insurance-react-ui/src/models/announcements/v1/Announcement.ts
import { Session } from "../../users/sessions/v1/Session";

export class Announcement {
    id?: number;
    createdDateTime?: string; //yyyy-mm-dd hh:mm:ss
    message?: string;
    announcementName?: string;
    session?: Session;

    constructor(obj?: any) {
        if (obj) {
            if (obj?.id) {
                this.id = obj.id;
            }
            if (obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if (obj?.message) {
                this.message = obj.message;
            }
            if (obj?.announcementName) {
                this.announcementName = obj.announcementName;
            }
            if (obj?.session) {
                this.session = new Session(obj.session);
            }
        }
    }
}